### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/mimalloc/actions/runs/4463465974#summary-12116021728

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/mimalloc/actions/runs/4463466056#summary-12116021920

### CODE REVIEW

File: doc/mimalloc-doc.h [hunk 1]
```diff
@@ -927,7 +927,7 @@ template<class T> struct mi_stl_allocator { }
 
 /*! \page build Building
 
-Checkout the sources from Github:
+Checkout the sources from GitHub:
```
Comment (jsoref) [inline doc/mimalloc-doc.h:930]: brand

File: docs/build.html [hunk 2]
```diff
@@ -98,7 +98,7 @@
 <div class="title">Building </div>  </div>
 </div><!--header-->
 <div class="contents">
-<div class="textblock"><p>Checkout the sources from Github: </p><div class="fragment"><div class="line">git clone https:<span class="comment">//github.com/microsoft/mimalloc</span></div>
+<div class="textblock"><p>Checkout the sources from GitHub: </p><div class="fragment"><div class="line">git clone https://github.com/microsoft/mimalloc</div>
```
Comment (jsoref) [inline docs/build.html:101]: I'm uncertain about why the clone url was corrupted, but it seems undesirable.

File: src/init.c [hunk 3]
```diff
@@ -568,7 +568,7 @@ static void mi_detect_cpu_features(void) {
   // FSRM for fast rep movsb support (AMD Zen3+ (~2020) or Intel Ice Lake+ (~2017))
   int32_t cpu_info[4];
   __cpuid(cpu_info, 7);
-  _mi_cpu_has_fsrm = ((cpu_info[3] & (1 << 4)) != 0); // bit 4 of EDX : see <https ://en.wikipedia.org/wiki/CPUID#EAX=7,_ECX=0:_Extended_Features>
+  _mi_cpu_has_fsrm = ((cpu_info[3] & (1 << 4)) != 0); // bit 4 of EDX : see <https://en.wikipedia.org/wiki/CPUID#EAX=7,_ECX=0:_Extended_Features>
```
Comment (jsoref) [inline src/init.c:571]: This corruption also seems undesirable

File: src/options.c [hunk 4]
```diff
@@ -507,7 +507,7 @@ static bool mi_getenv(const char* name, char* result, size_t result_size) {
   return (len > 0 && len < result_size);
 }
 #elif !defined(MI_USE_ENVIRON) || (MI_USE_ENVIRON!=0)
-// On Posix systemsr use `environ` to acces environment variables
+// On Posix systemsr use `environ` to access environment variables
```
Comment (jsoref) [inline src/options.c:510]: `acces` is often a posix thing, but, here it appears to just be a typo

File: test/test-stress.c [hunk 5]
```diff
@@ -7,7 +7,7 @@ terms of the MIT license.
 /* This is a stress test for the allocator, using multiple threads and
    transferring objects between threads. It tries to reflect real-world workloads:
    - allocation size is distributed linearly in powers of two
-   - with some fraction extra large (and some extra extra large)
+   - with some fraction extra large (and some extra, extra, large)
```
Comment (jsoref) [inline test/test-stress.c:10]: these commas are a grammatical thing which coincidentally mean that the duplicate word detection doesn't complain...

**Review by jsoref — COMMENTED** [review]
Most spelling corrections suggestions automatically suggested by Google Sheets. All fault mine

**Comment by daanx** [comment]
Thanks so much! I went through it and applied manually since the source diverged so much.

**Pull request closed without merging**